add documentation folder and exemplify custom abi

moshmage opened the pull request:
[dev-283](https://taikai.atlassian.net/jira/software/c/projects/DEV/boards/45?modal=detail&selectedIssue=DEV-283)

hvasconcelos commented on docs/custom-abi.md:
```diff
@@ -0,0 +1,77 @@
+# Using @taikai/dappkit with a custom ABI
```
with a custom smart contract ????

hvasconcelos commented on docs/custom-abi.md:
````diff
@@ -0,0 +1,77 @@
+console.log('Contract deployed; Address: ', tx.contractAddress);
+```
+
+### Using a custom contrat
````
contrat typo

moshmage commented:
> Would not be better to have separate instructions to deploy a custom smart contract from the instructions to interact with a custom smart contract ?

I've added a index of sorts to the custom-abi file, this should suffice to clear the doubts -- having two different files would leave both of them too empty or too repetitive -- but we can add more spacing between them if needed. Please recheck :)

In the meanwhile, I've added the "how to connect" as well as it was small enough
